src/generate-from-template.js

```diff
@@ -57,11 +57,13 @@ const generateFromTemplate = async ({ templateName }) => {
     }
   }
 
+  const regExpression = templatePackage.expression;
```

Comment:
instead of a regex why not take a function ? there might be more advanced cases than a regex could handle. the function could pull in data and validate against that etc

Reply:
agreed. there also needs to be a more descriptive name for any option that describes what the option is for rather than the type